apps/box/src/stores/useSettingsStore.ts

```diff
@@ -0,0 +1,56 @@
+import create, { StateCreator } from 'zustand';
+import { persist } from 'zustand/middleware';
+import AsyncStorage from '@react-native-async-storage/async-storage';
+import { useColorScheme } from 'react-native';
+
+export type ColorScheme = 'light' | 'dark';
+interface ModeSlice {
+  _hasHydrated: boolean;
+  setHasHydrated: (isHydrated: boolean) => void;
+  getMode: () => ColorScheme;
+  isAuto: boolean;
+  colorScheme: ColorScheme;
+  toggleIsAuto: () => void;
+  setColorScheme: (colorScheme: ColorScheme) => void;
+}
+const createModeSlice: StateCreator<
+  ModeSlice,
+  [],
+  [['zustand/persist', ModeSlice]],
+  ModeSlice
+> = persist(
+  (set, get) => ({
+    _hasHydrated: false,
+    setHasHydrated: (isHydrated) => {
+      set({
+        _hasHydrated: isHydrated,
+      });
+    },
+    isAuto: true,
+    colorScheme: 'dark',
+    setColorScheme: (colorScheme: ColorScheme) =>
+      set(() => ({ colorScheme: colorScheme })),
+    toggleIsAuto: () => set((state) => ({ isAuto: !state.isAuto })),
+    getMode: () => {
+      const { isAuto, colorScheme } = get();
+      const systemColorScheme = useColorScheme(); // eslint-disable-line react-hooks/rules-of-hooks
```

Comment:
Is this safe to use the hook here?

Reply:
Yes, so my understanding is that this is fine as long as you don't break an of the hooks rules. i.e don't call this function outside of a functional component or don't conditionally render it. But I also looked into it just to confirm it wasn't an issue. Came across [this](https://github.com/pmndrs/zustand/issues/42#issuecomment-695778068) from the creator of Zustand... They don't document it to not confuse users.

Reply:
Is it clearer to use `Appearance.getColorScheme()` here?

Reply:
Yes I could but I don't think `Appearance.getColorScheme()` subscribes to color scheme updates. So it won't be dynamic. Hence the reason for the hook.